implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringNode.java

```diff
@@ -797,12 +821,35 @@ void updateState(final State new_state, final Exception e) {
     if (state == null) {
       LOG.error("No state found. Maybe we need to stop?");
       return;
+    } else if (state.state == State.COMPLETE && 
+        (DateTime.currentTimeMillis() / 1000) - state.lastUpdateTime < 120) {
+      LOG.warn("Already complete?!?!");
+      return;
     }
     
     state.state = new_state;
     state.lastUpdateTime = DateTime.currentTimeMillis() / 1000;
     state.exception = e == null ? "" : e.getMessage();
-    cache.setState(cache_key, state, 300_000); // TODO config  
+    cache.setState(cache_key, state, 300_000); // TODO config
+    
+    AnomalyPredictionState cas = cache.getState(cache_key);
+    int cnt = 0;
+    while (!cas.equals(state) && cnt++ < 5) { 
+      LOG.error("Whoops? Failed cas?: " + cas);
+      try {
+        Thread.sleep(50);
+      } catch (InterruptedException e1) {
+        LOG.error("Unexpected interruption", e1);
+        return;
+      }
+      cache.setState(cache_key, state, 300_000); // TODO config
+      cas = cache.getState(cache_key);
+    }
+    if (cas.equals(state)) {
```

Comment:
If another thread updates the cache just  after #845  and before #846, `cas` won't be equal to `state`. If we are just logging, we should be ok.

Reply:
yeah just logging